Mage/src/main/java/mage/cards/ExpansionSet.java

```diff
@@ -171,7 +175,36 @@ public String toString() {
 
         return theBooster;
     }
+    
+    protected int PartnerCheck(List<Card> booster, boolean partnerAllowed, int max, int i){
+
+        for (Ability ability:booster.get(booster.size() - 1).getAbilities()){
+                    //Check if fetched card has the PartnerWithAbility
+                    if (ability instanceof PartnerWithAbility) {               
+                        //Check if the pack already contains a partner pair
+                        if (partnerAllowed){
+                            //Added card always replaces an uncommon card
+                            Card card = CardRepository.instance.findCard(((PartnerWithAbility) ability).getPartnerName()).getCard();
+                            if (i<max){
+                                booster.add(card);
+                            }
+                            else{
+                                booster.set(0, card);
+                            }
+                            //2 return value indicates found partner
+                            return 2;
+                        }
 
+                        else{
+                            //If partner already exists, remove card and loop again
```

Comment:
comment says : loop again, but the scope returns a value.

Reply:
You are right in that the comment might be a bit misleading as the looping technically happens outside the function. Essentially the returned value indicates that the maximum number of allowed partners is reached and thus another card must be generated.